eth/downloader/downloader.go

```diff
@@ -580,14 +594,14 @@ func (d *Downloader) spawnSync(fetchers []func() error) error {
 			break
 		}
 	}
-
 	d.queue.Close()
-	d.cancel()
-	wg.Wait()
+	d.Cancel()
```

Comment:
Here and just below is where the downloader gets to actually be aware of it's own cancel and waiting state.